### ISSUE TITLE
Issues when inserting data with a foreign key pointing back to the record being inserted.

### ISSUE THREAD

**Issue by ac931274** [zzzprojects/nmemory#24]
Hello there,
We are using Effort to run some tests and using CsvDataLoader to load the test data.  One of the tables is called User,  this table has an id (key), username and also a foreign key pointing back to the id of the user who modified the records last.  When we try and load the table we are getting the exception 
Effort.Exceptions.EffortException: Unhandled exception while trying to initialize the content of 

'Table<User>' table ---> NMemory.Exceptions.MultipleUniqueKeyFoundException: Unique key violation (Id). Error code: GenericError
    at NMemory.Indexes.UniqueIndex`2.Insert(TEntity item)
   at Effort.Internal.DbManagement.Engine.ExtendedTable`2.Initialize(IEnumerable`1 entities)
   at Effort.Internal.Common.DatabaseReflectionHelper.WrapperMethods.InitializeTableData[TEntity](ITable`1 table, IEnumerable`1 entities)
 --- End of inner exception stack trace ---
    at Effort.Internal.Common.DatabaseReflectionHelper.InitializeTableData(ITable table, IEnumerable`1 entities)
   at Effort.Internal.DbManagement.DbContainer.Initialize(DbSchema schema)
   at Effort.Provider.EffortProviderServices.<>c__DisplayClass4.<DbCreateDatabase>b__3(DbConnection x)
   at Effort.Provider.EffortProviderServices.Wrap[T](DbConnection connection, Func`2 action)
   at System.Data.Entity.Internal.DatabaseOperations.Create(ObjectContext objectContext)
   at System.Data.Entity.Internal.DatabaseCreator.CreateDatabase(InternalContext internalContext, Func`3 createMigrator, ObjectContext objectContext)
   at System.Data.Entity.Database.Create(DatabaseExistenceState existenceState)
   at System.Data.Entity.CreateDatabaseIfNotExists`1.InitializeDatabase(TContext context)
   at System.Data.Entity.Internal.InternalContext.PerformInitializationAction(Action action)
   at System.Data.Entity.Internal.InternalContext.PerformDatabaseInitialization()
   at System.Data.Entity.Internal.RetryAction`1.PerformAction(TInput input)
   at System.Data.Entity.Internal.LazyInternalContext.InitializeDatabaseAction(Action`1 action)
   at System.Data.Entity.Database.Initialize(Boolean force)

Is there anyway, without changing the entity model to load this data into Effort / NMemory?

**Comment by JonathanMagnan** [comment]
Hello @ac931274 ,

Do you think you could provide us a test project with this issue?

It will make easier/faster for my developer to getting started for investigating it.

Best Regards,

Jonathan

**Comment by ac931274** [comment]
I'll see what I can do for you

**Comment by ac931274** [comment]
Hi there,
I have had a bit of a dig and my apologies are due.  Your code is working as expected.  The issue was that when I used the CSV Extraction code on my database which had the tables [dbo].[users] & [history].[users] the second table having a different primary key to the first (since it stored the history of the first table) overwrote the first table with the file name "users.csv".  When the CSVLoader loaded this in, this is when the exception was thrown since duplicate primary keys were inserted.
I am planning on submitting a pull request for the fix I have made in my local version of the CSV Tool to append the schema name to the file if there are more than 1 schemas used.

Sorry again for wasting your time.

Jonathan
